Tests/FunctionalTests/Edge+PublicAPITests.swift

```diff
@@ -0,0 +1,140 @@
+//
+
+import AEPCore
+import AEPEdge
+import AEPEdgeIdentity
+import AEPServices
+import XCTest
+
+class EdgePublicAPITests: FunctionalTestBase {
+    private let exEdgeInteractProdUrlLocHint = URL(string: FunctionalTestConst.EX_EDGE_INTERACT_PROD_URL_STR_OR2_LOC)! // swiftlint:disable:this force_unwrapping
+    private let responseBody = "{\"test\": \"json\"}"
+
+    public class override func setUp() {
+        super.setUp()
+        FunctionalTestBase.debugEnabled = true
+    }
+
+    override func setUp() {
+        super.setUp()
+        continueAfterFailure = true
+        FileManager.default.clearCache()
+
+        // hub shared state update for 1 extension versions (InstrumentedExtension (registered in FunctionalTestBase), IdentityEdge, Edge) IdentityEdge XDM, Config, and Edge shared state updates
+        setExpectationEvent(type: FunctionalTestConst.EventType.HUB, source: FunctionalTestConst.EventSource.SHARED_STATE, expectedCount: 4)
+
+        // expectations for update config request&response events
+        setExpectationEvent(type: FunctionalTestConst.EventType.CONFIGURATION, source: FunctionalTestConst.EventSource.REQUEST_CONTENT, expectedCount: 1)
+        setExpectationEvent(type: FunctionalTestConst.EventType.CONFIGURATION, source: FunctionalTestConst.EventSource.RESPONSE_CONTENT, expectedCount: 1)
+
+        // wait for async registration because the EventHub is already started in FunctionalTestBase
+        let waitForRegistration = CountDownLatch(1)
+        MobileCore.registerExtensions([Identity.self, Edge.self], {
+            print("Extensions registration is complete")
+            waitForRegistration.countDown()
+        })
+        XCTAssertEqual(DispatchTimeoutResult.success, waitForRegistration.await(timeout: 2))
+        MobileCore.updateConfigurationWith(configDict: ["edge.configId": "12345-example"])
+
+        assertExpectedEvents(ignoreUnexpectedEvents: false)
+        resetTestExpectations()
+    }
+
+    func testSetLocationHint_sendEvent_sendsNetworkRequestWithLocationHint() {
+        setExpectationNetworkRequest(url: FunctionalTestConst.EX_EDGE_INTERACT_PROD_URL_STR_OR2_LOC, httpMethod: HttpMethod.post, expectedCount: 1)
+
+        Edge.setLocationHint("or2")
+        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"],
+                                              data: ["data": ["test": "data"]])
+        Edge.sendEvent(experienceEvent: experienceEvent)
+
+        // verify
+        assertNetworkRequestsCount()
+    }
+
+    func testSetLocationHint_withNilHint_sendEvent_sendsNetworkRequestWithoutLocationHint() {
+        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"],
+                                              data: ["data": ["test": "data"]])
+
+        setExpectationNetworkRequest(url: FunctionalTestConst.EX_EDGE_INTERACT_PROD_URL_STR_OR2_LOC, httpMethod: HttpMethod.post, expectedCount: 1)
+        Edge.setLocationHint("or2")
```

Comment:
nit: Do we want to extract or2 as a constant since we have the url for it in a constant.